### PR TITLE
Added Overwrite tests for F-71a with comments addressed

### PR DESCRIPTION
**Pull request by aditipande96**
Signed-off-by: Aditi Pande <aditi.pande@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by kachhwahadivya — APPROVED** [review]
PLease address codacy

File: tests/csm/rest/test_capacity_quota.py [hunk 1]
```diff
@@ -2670,3 +2670,109 @@ def test_43958(self):
...
+        max_size = test_cfg["max_size"]
+        max_objects_1 = test_cfg["max_objects_1"]
+        max_objects_2 = test_cfg["max_objects_2"]
+        self.log.info("Step 1: Set max_size and max_objects quota for user")
+        payload = self.csm_obj.iam_user_quota_payload(enabled, max_size, max_objects_1,
+                                                       check_on_raw=True)
+        result, resp = self.csm_obj.verify_get_set_user_quota(self.user_id, payload,
+                                                              verify_response=True)
+        assert result, resp
+        self.log.info("Step 2: Try uploading 1 object(obj-1) of max_size: %s ", max_size)
```
Comment (RajeshPratapSInghChouhan) [inline tests/csm/rest/test_capacity_quota.py:2695]: this will be successful, so can we remove "Try" from this ?

Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:2695]: you mean from msg?

Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: done

**Review by RajeshPratapSInghChouhan — APPROVED** [review]

File: tests/csm/rest/test_capacity_quota.py [hunk 2]
```diff
@@ -2670,3 +2670,111 @@ def test_43958(self):
...
+        assert result, resp
+        random_size = self.csm_obj.random_gen.randrange(1, max_size)
+        remaining_size = max_size - random_size
+        size_list = [random_size, remaining_size]
+        self.log.info("Get sorted list of random and remaining size")
+        size_list.sort()
+        self.log.info("Step 2: Try uploading 1 object(obj-1) of less than max_size"
+                      "and remaining size")
+        for size in size_list:
+            self.log.info("Uploading %s size object", size)
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:2747]: Overwrite

File: tests/csm/rest/test_capacity_quota.py [hunk 3]
```diff
@@ -2670,3 +2670,111 @@ def test_43958(self):
...
+                                                      check_on_raw=True)
+        result, resp = self.csm_obj.verify_get_set_user_quota(self.user_id, payload,
+                                                              verify_response=True)
+        assert result, resp
+        random_size = self.csm_obj.random_gen.randrange(1, max_size)
+        remaining_size = max_size - random_size
+        size_list = [random_size, remaining_size]
+        self.log.info("Get sorted list of random and remaining size")
+        size_list.sort()
+        self.log.info("Step 2: Try uploading 1 object(obj-1) of less than max_size"
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:2744]: Step 2: Upload Object 1 Name: %s and overwrite loop

**Review by ravindrachoudhari — APPROVED** [review]

File: tests/csm/rest/test_capacity_quota.py [hunk 4]
```diff
@@ -2670,3 +2670,111 @@ def test_43958(self):
...
+                                          self.akey, self.skey, object_size=int(size/1024),
+                                          block_size="1K")
+            assert resp, f'Put object Failed for {self.obj_name}.'
+        self.log.info("Step 3: Try uploading 1 object(obj-1) of remaining size")
+        try:
+            resp = s3_misc.create_put_objects(self.obj_name, self.bucket,
+                                          self.akey, self.skey,
+                                 object_size=int(size_list[0]/1024), block_size="1K")
+        except ClientError as error:
+            self.log.info("Expected exception received %s", error)
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:2758]: Expected failure when Object 1 Name : %s Size :%s which is same as overwrite size is uploaded

File: tests/csm/rest/test_capacity_quota.py [hunk 5]
```diff
@@ -2670,3 +2670,111 @@ def test_43958(self):
...
+            assert error.response['Error']['Code'] == "QuotaExceeded", \
+                                      "Overwriting object passed"
+        obj_name = f'{self.obj_name_prefix}{time.perf_counter_ns()}'
+        self.log.info("Step 4: Try uploading one more object with different name %s and"
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:2762]: Step 2: Upload Object 2 Name: %s and size : %s which is overwrite size within max object count : %s

File: tests/csm/rest/test_capacity_quota.py [hunk 6]
```diff
@@ -2670,3 +2670,111 @@ def test_43958(self):
...
+                              "size %s", obj_name, size_list[0])
+        resp = s3_misc.create_put_objects(obj_name, self.bucket,
+                                          self.akey, self.skey,
+                                   object_size=int(size_list[0]/1024), block_size="1K")
+        assert resp, f'Put Object Failed for {obj_name}.'
+        obj_name_1 = f'{self.obj_name_prefix}{time.perf_counter_ns()}'
+        obj_list = [obj_name, obj_name_1]
+        self.log.info("Step 5: Try overwriting above object and then upload one more object")
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:2770]: Step 5: Trying to overwriting Object 2 Name : %s and uploading one more object.

File: tests/csm/rest/test_capacity_quota.py [hunk 7]
```diff
@@ -2670,3 +2670,111 @@ def test_43958(self):
...
+        for objs in obj_list:
+            try:
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:2772]: Trying to upload object : %s, obj

File: tests/csm/rest/test_capacity_quota.py [hunk 8]
```diff
@@ -2670,3 +2670,111 @@ def test_43958(self):
...
+                resp = s3_misc.create_put_objects(objs, self.bucket,
+                                          self.akey, self.skey,
+                                 object_size=int(size_list[0]/1024), block_size="1K")
+            except ClientError as error:
+                self.log.info("Expected exception received %s", error)
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_capacity_quota.py:2777]: Expected failure as Quota has exhausted.

File: tests/csm/rest/test_capacity_quota.py (hunk 7 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py:2777]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 6 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 5 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 4 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: done

File: tests/csm/rest/test_capacity_quota.py [hunk 9]
```diff
@@ -2670,3 +2670,111 @@ def test_43958(self):
...
+        size_list.sort()
+        self.log.info("Step 2: Try uploading 1 object(obj-1) of less than max_size"
+                      "and remaining size")
+        for size in size_list:
+            self.log.info("Uploading %s size object", size)
+            resp = s3_misc.create_put_objects(self.obj_name, self.bucket,
+                                          self.akey, self.skey, object_size=int(size/1024),
+                                          block_size="1K")
+            assert resp, f'Put object Failed for {self.obj_name}.'
+        self.log.info("Step 3: Try uploading 1 object(obj-1) of remaining size")
```
[the remark this answers is not in the document]
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 3 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: done

File: tests/csm/rest/test_capacity_quota.py (hunk 2 above)
Comment (aditipande96) [reply tests/csm/rest/test_capacity_quota.py]: done

**Pull request merged**